### PR TITLE
Feature/refactor

### PR DESCRIPTION
**Pull request by takesei**
## 概要
各実験データ毎にばらばらになってしまっていた, 複数のmuscle\d.ipynbを統合するとともに, コードの見直しを行った

## 方法
殆どが関数化, モジュール化されていなかったため, コードの再利用性を高めるよう全体を書き直した. その上で

- `get_data(dataset_dir, content, t_range)`の導入により特定ディレクトリ下のすべての"content"に含まれる実験データをcsvで読み取り, pandas.DataFrameで開けるようにした
- `preprocess(function, list)`の導入により, 上記`get_data`にて取得したデータに対してかんたんに前処理を行えるようにした
- `plot_history(history)`の導入により, 一行にaccuracy/lossのplotを出力させるようにした
- `##### CONST NUMBER  SETTING #####`部分にて事前にわかる定数情報を入力するようにすることで, プログラムに与える必要のある情報を一箇所で設定/確認できるようにした

リファクタリングの観点では上記部分の変更を主に行ったが, 一方で学習時に, Callbackを呼び出すことで学習率の変更, 精度向上時のパラメータ保存なども行うようにした.

## 今後の方針
一方でリファクタリングする上で懸念点が発生した

### 精度が思ったよりも出ない.
書き直した上で期待していた精度が出なかった. 学習グラフを見るに典型的な過学習が発生していた. コードは書き換えたがモデル自体は全く変更していないので, 前処理の再検討やモデル選択の見直し等を考えてみるべきかもしれない. 必須である

一方で[muscle3](https://github.com/Meg-Kato/myoelectricity/blob/6c4fbf8d2b/muscle3.ipynb)では同じモデルのはずなのに異なる結果(test acc100%)が出ていた. この違いはどこから来ているのだろうか

学習グラフを見てみると２つのグラフがあまりにも似すぎている. どこかでデータがコンタミしている可能性などデバッグをしながらコードの再確認が必要と思われる.

基本的にvalidationとtrainの正答率は離れて当然(dropoutやL1が抜けるため)である([c.f.](https://keras.io/ja/getting-started/faq/#training-losstesting-loss))ため, もし本コードを使わなかった場合があったとしても見直しは必須であろう

### CODE REVIEW

**Comment by takesei** [comment]
そういえばなかったのでREADMEも追加します

**Comment by Meg-Kato** [comment]
Thank you so much.
